File: frontend/src/api/ai-config.ts
import http from './http'

// ---------- Types ----------

export type AIProvider = 'openai' | 'claude' | 'custom'

export interface AIConfigCreate {
  name: string
  provider: AIProvider
  model: string
  api_key: string
  base_url?: string
  is_default?: boolean
}

export interface AIConfigUpdate {
  name?: string
  provider?: AIProvider
  model?: string
  api_key?: string
  base_url?: string
  is_default?: boolean
}

export interface AIConfigItem {
  id: string
  name: string
  provider: AIProvider
  model: string
  api_key_masked: string
  base_url: string | null
  is_default: boolean
  created_at: string
  updated_at: string
}

export interface AIConfigTestPayload {
  provider: AIProvider
  model: string
  api_key: string
  base_url?: string
}

export interface AIConfigTestResult {
  ok: boolean
  message?: string
  latency_ms?: number
}

// ---------- API ----------

export async function getAIConfigs(): Promise<AIConfigItem[]> {
  const res = await http.get('/ai-configs')
  return res.data
}

export async function createAIConfig(payload: AIConfigCreate): Promise<AIConfigItem> {
  const res = await http.post('/ai-configs', payload)
  return res.data
}

export async function updateAIConfig(id: string, payload: AIConfigUpdate): Promise<AIConfigItem> {
  const res = await http.patch(`/ai-configs/${id}`, payload)
  return res.data
}

export async function deleteAIConfig(id: string): Promise<void> {
  await http.delete(`/ai-configs/${id}`)
}

export async function testAIConfig(payload: AIConfigTestPayload): Promise<AIConfigTestResult> {
  const res = await http.post('/ai-configs/test', payload)
  return res.data
}

File: backend/schemas/ai_config.py
from datetime import datetime
from typing import Literal
import uuid

from pydantic import BaseModel, field_validator


ProviderType = Literal["openai", "claude", "custom"]


class AIConfigTest(BaseModel):
    provider: ProviderType
    model: str
    api_key: str
    base_url: str | None = None


class AIConfigTestResult(BaseModel):
    ok: bool
    message: str | None = None
    latency_ms: int | None = None


class AIConfigCreate(BaseModel):
    name: str
    provider: ProviderType = "openai"
    model: str
    api_key: str
    base_url: str | None = None
    is_default: bool = False

    @field_validator("name")
    @classmethod
    def name_not_empty(cls, v: str) -> str:
        if not v.strip():
            raise ValueError("配置名称不能为空")
        return v.strip()

    @field_validator("model")
    @classmethod
    def model_not_empty(cls, v: str) -> str:
        if not v.strip():
            raise ValueError("模型名称不能为空")
        return v.strip()


class AIConfigUpdate(BaseModel):
    name: str | None = None
    model: str | None = None
    api_key: str | None = None
    base_url: str | None = None
    is_default: bool | None = None


class AIConfigResponse(BaseModel):
    id: uuid.UUID
    name: str
    provider: str
    model: str
    # API Key 脱敏：前4位 + **** + 后4位，不返回明文
    api_key_masked: str
    base_url: str | None
    is_default: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
